ReactAndroid/src/main/java/com/facebook/react/modules/toast/ToastModule.java

```diff
@@ -46,7 +56,30 @@ public String getName() {
   }
 
   @ReactMethod
-  public void show(String message, int duration) {
-    Toast.makeText(getReactApplicationContext(), message, duration).show();
+  public void show(String message, int duration, final int id) {
+    Toast toast = Toast.makeText(getReactApplicationContext(), message, duration);
+    toast.show();
+    mToasts.put(id, toast);
+    new Timer().schedule(new TimerTask() {
```

Comment:
That was my first thought, but then I wasn't sure when it'll get removed from the map since I'm not keeping any reference to the toast object.